CMakeLists.txt

```diff
@@ -349,8 +350,14 @@ endif()
 
 # expat
 if(NOT DISABLE_XML)
-	find_package(EXPAT)
-	target_link_libraries(lcf EXPAT::EXPAT)
+	find_package(expat)
+	if(NOT expat_FOUND)
+		# Fallback to old expat detection
+		find_package(EXPAT REQUIRED)
+		target_link_libraries(lcf EXPAT::EXPAT)
+	else()
+		target_link_libraries(lcf expat::expat)
+	endif()
```

Comment:
Flipping the order of branches would reduce cognitive load of this code by a bit. Same again further down one more time.